### ISSUE TITLE
useDocument and useCollection in the same component causes useCollection to return an empty array

### ISSUE THREAD

**Issue by rolznz** [nandorojo/swr-firestore#105]
This causes the collection to return an empty array
`
const { data: document } = useDocument<IChallenge>('challenges/challenge-1');
const { data: collection } = useCollection<IChallenge>('challenges');
console.log('document', document, 'collection', collection);
`
Result:
`document {id: 'abc'…},  collection []`

However, the following seems to work fine (just swapping around the calls)
`
const { data: collection } = useCollection<IChallenge>('challenges');
const { data: document } = useDocument<IChallenge>('challenges/challenge-1');
console.log('document', document, 'collection', collection);
`
Result:
`document {id: 'abc'…}, collection (8) [{…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}]`

**Comment by rolznz** [comment]
Note: I actually use a filter on the collection query, so there is a valid usecase for this. I have just made the simplest way to reproduce the issue.

**Comment by solkpolk** [comment]
`const {data:user, error:errorUser} = useDocument(`users/${uid}`, {listen:false})
    const {data:niftys} = useCollection("nifty", {
        where: ['userId', "==", uid],
        listen: true
    })`

**Comment by rolznz** [comment]
> `const {data:user, error:errorUser} = useDocument(`users/${uid}`, {listen:false}) const {data:niftys} = useCollection("nifty", { where: ['userId', "==", uid], listen: true })`

I don't think this is the solution. I shouldn't be forced to listen to changes.

**Comment by 345ml** [comment]
I am suffering from the same phenomenon.
In my case, it seems to be happening in Android cases where the where condition, etc. is changed dynamically.

If I revalidate it after some time, it goes away, but it seems to need about 5 seconds to go by.

**Comment by rolznz** [comment]
@345ml 

I have started a complete rewrite of this library as it is no longer maintained and there are many bugs. If you are interested it's here: https://github.com/rolznz/swr-firestore (Note that the syntax is different in order to make use of the latest version of firebase)

**Comment by 345ml** [comment]
@rolznz 
Alright.
Good job.
